### PR TITLE
BRW-47: refactor both methods "getBean()" in "beanFactory"

### PR DESCRIPTION
**Pull request by AlexandrNavozenko**
### Issue/Task

_[ List of issues or tasks in the form Task-xxx, Issue-xxx.]_

### Description

_[ Description of the changes for this PR.]_

### Testing

_[ Description of what testing was done. ]_

### Related Reviews

_[ Links to other PRs]_

### Screenshots

_[ Put your diagrams or screenshots with some highlights]_

### CODE REVIEW

File: src/main/java/svydovets/core/context/beanDefinition/BeanDefinitionFactory.java [hunk 1]
```diff
@@ -139,8 +141,8 @@ private String doResolveScopeName(Scope scopeAnnotation) {
 
             throw new UnsupportedScopeException(errorMessage);
         }
-        log.trace("Scope values is not specified explicitly. Set '{}' by default", ApplicationContext.SCOPE_SINGLETON);
-        return ApplicationContext.SCOPE_SINGLETON;
+        log.trace("Scope values is not specified explicitly. Set '{}' by default", SCOPE_SINGLETON);
+        return SCOPE_SINGLETON;
```
Comment (Pan14ek) [inline src/main/java/svydovets/core/context/beanDefinition/BeanDefinitionFactory.java:145]: What do you think about the Scope enum? Is it possible to use it?

File: src/main/java/svydovets/core/context/beanFactory/BeanFactory.java [hunk 2]
```diff
@@ -112,21 +117,21 @@ private Object postProcessAfterInitialization(Object bean, String beanName) {
...
         return postProcessAfterInitialization(bean, beanName);
     }
 
-    private void postConstructInitialization(Object bean) {
+    private void postConstructInitialization(Object bean) throws NoUniquePostConstructException {
         Class<?> beanType = bean.getClass();
         Method[] declaredMethods = beanType.getDeclaredMethods();
         Predicate<Method> isAnnotatedMethod = method -> method.isAnnotationPresent(PostConstruct.class);
 
         // todo: Refactor a little bit. Collect methods to list to avoid second filtering
```
Comment (Pan14ek) [inline src/main/java/svydovets/core/context/beanFactory/BeanFactory.java:129]: Can we remove this `todo`?

**Review by Pan14ek — COMMENTED** [review]
PTAL

File: src/main/java/svydovets/core/context/beanDefinition/BeanDefinitionFactory.java (hunk 1 above)
Comment (AlexandrNavozenko) [reply src/main/java/svydovets/core/context/beanDefinition/BeanDefinitionFactory.java:145]: on first iteration I added enum, but I had problem when init default value for @Scope

File: src/main/java/svydovets/core/context/beanFactory/BeanFactory.java (hunk 2 above)
Comment (AlexandrNavozenko) [reply src/main/java/svydovets/core/context/beanFactory/BeanFactory.java:129]: This should be fixed in another task. The scope of this task did not include

**Review by Pan14ek — APPROVED** [review]
LGTM

**Pull request merged**